test/integration/kubernetes_deploy_test.rb

```diff
@@ -635,6 +635,19 @@ def test_bad_container_on_daemon_sets_fails
     ], in_order: true)
   end
 
+  def test_resource_quotas_are_deployed_first
+    forced_timeout = 10
+    KubernetesDeploy::Deployment.any_instance.stubs(:timeout).returns(forced_timeout)
+    result = deploy_fixtures("resource-quota")
+    assert_deploy_failure(result)
+    assert_logs_match_all([
+      "Predeploying priority resources",
+      "Deploying ResourceQuota/resource-quotas (timeout: 30s)",
+      "Deploying all resources",
```

Comment:
I'd suggest replacing this assertion with one for `ResourceQuota/resource-quotas                     Available` (so this test also proves the quota part of the deploy succeeded), and adding an additional assertion at the end on the event to show that the web failure had to do with the quota. Otherwise, the test could pass because of the forced short timeout even if the web rollout were eventually going to succeed.

Sidenote: it sucks that this deploy times out, since we should be able to tell right away that it is doomed. Would you be willing to explore (as a follow-up, not in this PR) making the deployment itself fail fast on observing the related status condition?
```yaml
    - lastTransitionTime: 2017-09-25T17:04:07Z
      lastUpdateTime: 2017-09-25T17:04:07Z
      message: 'unable to create pods: pods "web-3393528609-" is forbidden: failed
        quota: resource-quotas: must specify limits.cpu,limits.memory,requests.cpu,requests.memory'
      reason: FailedCreate
      status: "True"
      type: ReplicaFailure
```